tasties_app/tests/test_category.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+from django.contrib.auth.models import User
+from tasties_app.models import Category, Recipe
+from django.db import IntegrityError
+from django.core.exceptions import ValidationError
+
+
+@pytest.fixture()
+def recipes():
+    """
+    This fixture generates some test Recipes
+
+    Returns:
+        tuple<Recipe>: A number of Recipe test objects
+    """
+    user = User.objects.create_user("testuser1", "password")  # This is required to create Recipes
+    recipe_1 = Recipe(
+        title="Test Recipe1",
+        author_id=user,
+        description="Test Description1",
+        directions="Test Directions1",
+        minutes_to_make=1,
+        recipe_picture="test_picture1",
+    )
+    recipe_1.save()
+    recipe_2 = Recipe(
+        title="Test Recipe2",
+        author_id=user,
+        description="Test Description2",
+        directions="Test Directions2",
+        minutes_to_make=1,
+        recipe_picture="test_picture2",
+    )
+    recipe_2.save()
+    return (recipe_1, recipe_2)
+
+
+@pytest.fixture()
+def categories():
+    """
+    This fixture generates some test Categories
+
+    Returns:
+        tuple<Category>: A number of Category test objects
+    """
+    category_1 = Category.objects.create(category_name="Breakfast")
+    category_2 = Category.objects.create(category_name="Mushrooms")
+    category_3 = Category.objects.create(category_name="Beef")
+    return (category_1, category_2, category_3)
+
+
+@pytest.fixture()
+def categorized_recipes(recipes, categories):
+    """
+    This fixture assigns Categories to Recipes
+
+    Args:
+        recipes (fixture): fixture 'recipes'
+        categories (fixture): fixture 'categories'
+
+    Returns:
+        tuple<tuple<Recipe>, tuple<Category>>: The Recipes generated by 'recipes' assigned with
+                       Categories generated by 'categories', as well as the Categories themselves
+    """
+    recipes[0].categories.add(categories[0])
+    recipes[0].categories.add(categories[1])
+    recipes[1].categories.add(categories[1])
+    return recipes, categories
+
+
+class TestCategoryModel:
+    @pytest.mark.django_db
+    def test_create_categories(self, categories, recipes):
+        """
+        Verifies fixtures 'categories' and 'recipes' properly
+           saved their data as expected
+
+        Args:
+            categories (fixture): fixture 'categories'
+            recipes (fixture): fixture 'recipes'
+        """
+        all_categories = Category.objects.all()
+        all_recipes = Recipe.objects.all()
+        for category in categories:
+            assert category in all_categories
```

Comment:
What is this test for? if you go over categories, doesn't `category in all_categories` every time?

Reply:
Please notice that this test is for verifying that the `Category` objects returned by the fixture `categories` (ditto with `recipes`) were actually saved to the database. If they're not actually saved properly, then the assertion will fail, because we will not find said `Category` objects returned by the `categories` fixture in `Category.objects.all()`.

Reply:
If the categories aren't saved, the `all` method will return an empty query set. Will the assertion fail in this case?

Reply:
> > If the categories aren't saved, the `all` method will return an empty query set. Will the assertion fail in this case?
> 
> Yes, since we're trying to assert that a tangible, existing `Category` object is in an empty set. I expect this to be the behavior. However, I will now test out if this is in fact the case,

See these changes to the code:

![image](https://user-images.githubusercontent.com/116169213/208728907-2f68134b-91a3-469e-8504-392b203136bc.png)

![image](https://user-images.githubusercontent.com/116169213/208729038-6e552d0d-192c-414e-9a62-5db5288cd339.png)

I commented out the 'create'/'save' options, and ran `test_create_categories`. I got this result:

[test_create_categories_log.txt](https://github.com/redhat-beyond/tasties/files/10270814/test_create_categories_log.txt)